python/templates/Object.h.jinja2

```diff
@@ -31,6 +32,9 @@ class {{ class.bare_type }} {
   friend class {{ class.bare_type }}CollectionIterator;
 
 public:
+  using mutable_type = Mutable{{ class.bare_type }};
+  using collection_type = {{ class.bare_type }}Collection;
```

Comment:
Thinking about this here, I think the `Mutable` objects should also have at least the `collection_type`. Mainly for symmetry reasons and also because it would allow a (cumbersome) way to come back from the `Mutable` types to the default ones.

Reply:
Done!